test/linalg/bunchkaufman.jl

```diff
@@ -107,7 +107,7 @@ end
 
 # test example due to @timholy in PR 15354
 let
-    A = rand(6,5); A = A'*A
+    A = rand(6,5); A = complex(A'*A) # to avoid calling the real-lhs-complex-rhs method
```

Comment:
It works but the test is with `==` here, probably to make sure that exactly the same method is called. For lhs `Matrix{Float64}` and rhs `Vector{Complex{Float64}}` the we are using the special method that reinterprets the rhs to avoid complex arithmetic.